### DISCUSSION TITLE
How can I map a keyboard shortcut to "Add to dictionary"?

### DISCUSSION THREAD

**Discussion by zetashift** [valentjn/vscode-ltex#282] [Q&A] [answered]
Hiya! First off, thanks a lot for this amazing extension, writing docs is a lot more fun with this.
I did a quick peek through the docs, but couldn't find exactly what I was looking for:
I want to make a keybinding for `ltex.addToDictionary`, I'm using the Vim keybindings so I would like to use my mouse as little as possible. 
In `Keyboard Shortcuts` in VScode I made a keybinding to `ltex.addToDictionary` to `Z-G` but I get an error when I actually use it.:
`Running the contributed command: 'ltex.addToDictionary' failed.`

My question basically is what command is exactly run for adding it to the dictionary? When I add a word through the `Quick Fix` route it works fine.

**Comment by valentjn** [comment, marked as answer]
The usual way to execute a quick fix with the keyboard, such as adding a word to the dictionary, is `Ctrl+.` (Control and period keys), `Arrow-Down`, and `Return`.

`ltex.addToDictionary` is an internal (undocumented) command. All supported commands are documented [here](https://valentjn.github.io/vscode-ltex/docs/commands.html). The reason why `ltex.addToDictionary` doesn't work for you is that it expects a list of arguments (`uri` with the document URI and `words` with the list of words to add). When running the quick fix via the context menu, VS Code supplies the arguments because the language server part (LT<sub>E</sub>X LS) told it before what values to use in case it wants to run the quick fix. When running it via a keybinding, the arguments are missing, and the command fails.

I think VS Code is currently not exposing the list of quick fixes for a selection in the editor to allow extensions running one of them manually (e.g., by another command that has been triggered by a keyboard shortcut).

**Reply by valentjn** [reply]
The next release will hide these internal commands from the user interface to reduce future confusion.

**Reply by zetashift** [reply]
Thank you very much, I'll keep it in mind :D
